File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int,int> lost_m; 
        for(int i = 0; i<matches.size(); i++){
            int loser  = matches[i][1]; 
            lost_m[loser]++; 
        }

        vector <int> not_lost; 
        vector<int> lost_once; 

        for(int i = 0; i < matches.size(); i++){
            int loser = matches[i][1]; 
            int winner = matches[i][0]; 

            if(lost_m.find(winner)== lost_m.end()){
                not_lost.push_back(winner); 
                lost_m[winner] = 2; 
            }
            if(lost_m[loser] == 1){
                lost_once.push_back(loser); 
            }
            
        }
        sort(begin(lost_once), end(lost_once));
        sort(begin(not_lost), end(not_lost));
        return {not_lost, lost_once}; 
    }
};

File: 1354-find-players-with-zero-or-one-losses/find-players-with-zero-or-one-losses.java
import java.util.*;

class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Map<Integer, Integer> lost_m = new HashMap<>(); 
        
        // Count losses
        for(int i = 0; i < matches.length; i++){
            int loser = matches[i][1]; 
            lost_m.put(loser, lost_m.getOrDefault(loser, 0) + 1); 
        }
        
        List<Integer> not_lost = new ArrayList<>(); 
        List<Integer> lost_once = new ArrayList<>(); 
        
        for(int i = 0; i < matches.length; i++){
            int loser = matches[i][1]; 
            int winner = matches[i][0]; 

            // If winner is not in the map, add to not_lost and mark in map
            if(!lost_m.containsKey(winner)){
                not_lost.add(winner); 
                lost_m.put(winner, 2); 
            }
            // If loser has exactly 1 loss, add to lost_once
            if(lost_m.get(loser) == 1){
                lost_once.add(loser); 
            } 
        }
        
        Collections.sort(lost_once);
        Collections.sort(not_lost);
        
        return Arrays.asList(not_lost, lost_once); 
    }
}